adc/ad7779/imx6ull/gpio.c

```diff
@@ -0,0 +1,150 @@
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <sys/platform.h>
+
+#include <phoenix/arch/imx6ull.h>
+
+#include <imx6ull-gpio.h>
+
+#include "../ad7779.h"
+
+typedef struct {
+	int port;
+	int dir;
+} gpio_dev_t;
+
+struct {
+	gpio_dev_t reset;
+	gpio_dev_t start;
+} gpio;
+
+static int open_gpio(gpio_dev_t *gdev, int nr)
+{
+	unsigned tries = 100;
+	char port[16];
+	int res;
+
+	if (nr > 5 && nr < 1)
+		return -1;
+
+	res = snprintf(port, sizeof(port), "/dev/gpio%d/port", nr);
+	if (res < 0)
```

Comment:
actually for `snprintf`  `res >= sizeof(port)` means that we truncated the string